core/src/main/java/hudson/FilePath.java

```diff
@@ -1762,6 +1773,8 @@ public Void invoke(File f, VirtualChannel channel) throws IOException, Interrupt
                 try (InputStream fis = Files.newInputStream(reading(f).toPath());
                      OutputStream out = p.getOut()) {
                     org.apache.commons.io.IOUtils.copy(fis, out);
+                } catch (InvalidPathException e) {
```

Comment:
I am trying to ensure that any plugin code that was expecting an invalid path to report an `IOException` over `p.error(x)` will still get one, otherwise anyone using the other end of the input stream will get a `InvalidPathException` as the cause of the IOE from their read... an edge case, but given the assumptions plugins are making I felt it was worth it